### ISSUE TITLE
Multivariate resultants

### ISSUE THREAD

**Issue by VictoriaSchleis** [oscar-system/Oscar.jl#1260]
I would like an implementation of multivariate resultants in Oscar, analogous to the Singular implementation, compatible with polynomials over Flint fields. There is resultant(::MPoly, ::MPoly, ::Int), but it's not documented, and I'm not sure which fields it is compatible with.

**Comment by fieker** [comment]
On Wed, Apr 13, 2022 at 04:40:30AM -0700, VictoriaSchleis wrote:
> I would like an implementation of multivariate resultants in Oscar, analogous to the Singular implementation, compatible with polynomials over Flint fields. There is resultant(::MPoly, ::MPoly, ::Int), but it's not documented, and I'm not sure which fields it is compatible with. 
> 
Should work with all exact fields...

**Comment by tthsqe12** [comment]
The `resultant(::MPoly, ::MPoly, ::Int)` is in Hecke.jl and is not "the multivariate resultant" as is simply the univariate resultant w.r.t. a given variable.

**Comment by VictoriaSchleis** [comment]
Yes, that is precisely what I meant - I just meant it in the way that currently, the documented cases only work where the input is a univariate polynomial

**Comment by fingolfin** [comment]
@thofma also did #3546 some time ago.

@JohnAAbbott can you please get to the bottom of this

**Comment by jankoboehm** [comment]
See here

https://www.singular.uni-kl.de/Manual/4-4/sing_310.htm#SEC350

**Comment by JohnAAbbott** [comment]
@VictoriaSchleis You asked about this roughly 2.5 years ago.  Are you still interested?  Was the function pointed out by @tthsqe12 what you needed?

**Comment by fingolfin** [comment]
@JohnAAbbott  maybe try contacting @VictoriaSchleis directly to see if she is still interested?

**Comment by JohnAAbbott** [comment]
> [@JohnAAbbott](https://github.com/JohnAAbbott) maybe try contacting [@VictoriaSchleis](https://github.com/VictoriaSchleis) directly to see if she is still interested?

Email sent...

**Comment by VictoriaSchleis** [comment]
Hi all, sorry for the late reply. Only saw this when checkig my github. To be honest, it's been a few years and I can't remember what I needed multivariate resultants for. I still think it would be nice to have, but it isn't an immediate concern for me right now.

@JohnAAbbott, I don't have access to my old Tuebingen email address anymore, if you need to contact me please use either my IAS or my Durham address.

**Comment by fingolfin** [comment]
So the remaining question then is if we want to provide a nice OSCAR interface to  Singular's `mpresmat` (assuming this is the functionality meant in this issue; so far this is our best guess).

But this raised the question what this actually *does*, i.e. what is the "multipolynomial resultant matrix of the input system". Someone may have to look at the Singular implementation to figure that out (we need to put that kind of information in the documentation for our wrapper if we provide one).

Note that we also already have this:
```
  resultant(F::Vector{MPolyRingElem})

  Return the Macaulay resultant of a list of n homogeneous
  polynomials in n variables.
```
But this is something different (`mpresmat` for "multipolynomial resultant matrix" returns a `module` (used as a sparse matrix) while our `resultant` method returns a single polynomial.


@hannes14 will look into it.

**Comment by fingolfin** [comment]
@hannes14 did you make any progress on this?
